### PR TITLE
Added BetweenConditionBuilder

### LINKED ISSUE #11611: Query builder between 2 columns
Предлагаю фичу, ну или как минимум дописать документацию.

В документации:
http://www.yiiframework.com/doc-2.0/yii-db-query.html

Написано:

> between: operand 1 should be the column name, and operand 2 and 3 should be the starting and ending values of the range that the column is in. For example, ['between', 'id', 1, 10] will generate id BETWEEN 1 AND 10.

Можно ли сделать поддержку следующего?
$this->where(['between', $value, 'column_from', 'column_to'])->all();

Как я читал SQL подобную конструкцию поддерживает. MySQL поддерживает подобный синтаксис.

``` SQL
SELECT * FROM  `mytable` 
WHERE 4 
BETWEEN  `from` AND  `to`
```

### PR DESCRIPTION
**Pull request by SilverFire**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | Replaces #12678, Fixes #11611

### CODE REVIEW

File: framework/db/conditions/BetweenColumnsCondition.php [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+ *
+ * @author Dmytro Naumenko <d.naumenko.a@gmail.com>
+ * @since 2.0.14
+ */
+class BetweenColumnsCondition implements ConditionInterface
+{
+    /**
+     * @var string $operator the operator to use (e.g. `BETWEEN` or `NOT BETWEEN`)
+     */
+    protected $operator;
```
Comment (samdark) [inline framework/db/conditions/BetweenColumnsCondition.php]: Do we need these to be protected instead of private?

**Review by samdark — APPROVED** [review]

File: framework/db/conditions/BetweenColumnsCondition.php (hunk 1 above)
Comment (SilverFire) [reply framework/db/conditions/BetweenColumnsCondition.php]: Well, to be honest all properties is conditions may be private. They are not intended to be reused, except abstract class `ConjunctionCondition`. I can make all 'em private with a separate PR

Comment (samdark) [reply framework/db/conditions/BetweenColumnsCondition.php]: Why separate PR? Let's make em private, I think.

Comment (SilverFire) [reply framework/db/conditions/BetweenColumnsCondition.php]: Ok

**Pull request merged**